tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -4309,5 +4309,40 @@ llvm::Optional<Value> convertOneHotOp(PatternRewriter& rewriter, Operation* op,
       .getResult();
 }
 
+// Lowers 32-bit floating sign operator to a sequence of TOSA ops.
+llvm::Optional<Value> convertSignOp(PatternRewriter& rewriter, Operation* op,
+                                    Value input, RankedTensorType output_type) {
+  // Only support 32-bit floating point currently.
+  if (!output_type || !output_type.getElementType().isF32()) {
```

Comment:
Could you include the integer version of the operation as well? It shouldn't be that difficult to include and would allow us to fully support the operation.

Reply:
Sure thing. I add i32 version for tf.sign and tfl.sign, and keep i8/i16 unsupported as the real quantization hasn't yet supported.